Makefile

```diff
@@ -32,3 +34,23 @@ pull_translations:
 	git clean -fdX conf/locale/rtl
 	git clean -fdX conf/locale/eo
 	i18n_tool validate
+
+requirements: ## install development environment requirements
+	pip install -qr requirements/edx/development.txt --exists-action w
+
+upgrade: ## update the pip requirements files to use the latest releases satisfying our constraints
+	pip install -q pip-tools
+	pip-compile --upgrade -o requirements/edx/coverage.txt requirements/edx/coverage.in
+	pip-compile --upgrade -o requirements/edx/paver.txt requirements/edx/paver.in
+	pip-compile --upgrade -o requirements/edx-sandbox/shared.txt requirements/edx-sandbox/shared.in
+	pip-compile --upgrade -o requirements/edx-sandbox/base.txt requirements/edx-sandbox/base.in
+	pip-compile --upgrade -o requirements/edx/base.txt requirements/edx/base.in
+	pip-compile --upgrade -o requirements/edx/testing.txt requirements/edx/testing.in
+	pip-compile --upgrade -o requirements/edx/development.txt requirements/edx/development.in
```

Comment:
Are these all just in a specific order which guarantees that no compilation is performed with an old compiled .txt file?  In general I thought that no .in file would install a .txt file, but maybe we want to do that for performance reasons?

Reply:
Right, this sequence compiles them such that the ones which are included in other files get built before the ones that include them.  The reason for including ".txt" instead of ".in" files is to ensure consistency of dependency versions across different contexts (so pip-tools doesn't resolve to one version for production and a different one for tests after constraints from additional dependencies are imposed).  I'm currently making an edit to the draft OEP to explain that reasoning.

Reply:
There needs to be a comment here pointing at that explanation.  This looks easy to screw up.